shared/src/commonMain/kotlin/org/jetbrains/kotlinconf/screens/ScheduleScreen.kt

```diff
@@ -305,39 +315,105 @@ fun ScheduleList(
-                        lightning = session.isLightning,
-                        time = session.badgeTimeLine,
-                        timeNote = null,
-                        status = when {
-                            session.isFinished -> TalkStatus.Past
-                            session.isLive -> TalkStatus.Now
-                            else -> TalkStatus.Upcoming
-                        },
-                        onSubmitFeedback = { emotion ->
-                            onSubmitFeedback(session.id, emotion)
-                        },
-                        onSubmitFeedbackWithComment = { emotion, comment ->
-                            onSubmitFeedbackWithComment(session.id, emotion, comment)
-                        },
-                        onClick = { onSession(session.id) },
+                        onBookmark = onBookmark,
+                        onSubmitFeedback = onSubmitFeedback,
+                        onSubmitFeedbackWithComment = onSubmitFeedbackWithComment,
+                        onSession = onSession,
                         modifier = Modifier
                             .fillMaxWidth()
                             .padding(horizontal = 12.dp, vertical = 8.dp)
                     )
                 }
+
+                is ServiceEventItem -> {
+                    val event = item.value
+                    ServiceEvent(
+                        event = event,
+                        modifier = Modifier.padding(horizontal = 12.dp, vertical = 4.dp),
+                    )
+                }
             }
         }
     }
 }
+
+@Composable
+private fun SessionCard(
+    session: SessionCardView,
+    onBookmark: (SessionId, Boolean) -> Unit,
+    onSubmitFeedback: (SessionId, Emotion?) -> Unit,
+    onSubmitFeedbackWithComment: (SessionId, Emotion, String) -> Unit,
+    onSession: (SessionId) -> Unit,
+    modifier: Modifier = Modifier,
+    titleHighlights: List<IntRange> = emptyList(),
+    tagHighlights: List<String> = emptyList(),
+    speakerHighlights: List<IntRange> = emptyList(),
+) {
+    TalkCard(
+        title = session.title,
+        titleHighlights = titleHighlights,
+        bookmarked = session.isFavorite,
+        onBookmark = { isBookmarked -> onBookmark(session.id, isBookmarked) },
+        tags = session.tags,
+        tagHighlights = tagHighlights,
+        speakers = session.speakerLine,
+        speakerHighlights = speakerHighlights,
+        location = session.locationLine,
+        lightning = session.isLightning,
+        time = session.badgeTimeLine,
+        timeNote = session.startsInMinutes?.let { count ->
+            stringResource(Res.string.schedule_in_x_minutes, count)
+        },
+        status = when {
+            session.isFinished -> TalkStatus.Past
+            session.isLive -> TalkStatus.Now
+            session.isUpcoming -> TalkStatus.Upcoming
+            else -> TalkStatus.Upcoming // Shouldn't happen
```

Comment:
It looks like we need to log an error here

Reply:
Good note - replaced this whole mess with an enum so we always have a specific session state.